### ISSUE TITLE
Function reassignment keeps old units [bug]

### ISSUE THREAD

**Issue by asher-pembroke** [EnsembleGovServices/kamodo-core#52]
## Describe the bug
Function reassignment should update units of the new function. However, reassignment with a unit-less function does not remove the old function units from the signature registry or function metadata.

## To Reproduce
```python
def test_unit_reassignment():
    k = Kamodo()
    
    #  initialze a function with units
    k['f(x[cm])[cm]'] = 'x**2-x-1'
    
    # update the function so it has units
    k['f(x[m])[m]'] = 'x**2-x-1'
    
    assert k.f.meta['units'] == 'm'
    assert k.f.meta['arg_units']['x'] == 'm'
    assert k.signatures['f']['units'] == 'm'
    assert k.signatures['f']['arg_units']['x'] == 'm'
    
    # update function to remove units
    k['f'] = 'x**2-x-1' 
    assert k.f.meta['units'] == ''
    assert len(k.f.meta['arg_units']) == 0
    assert k.signatures['f']['units'] == ''
    assert len(k.signatures['f']['arg_units']) == 0
```

** The above test should be added to test_kamodo.py to validate changes related to this issue.**

**Comment by asher-pembroke** [comment]
@pythonguru101  Looks good! Open a PR for this one.

**Comment by asher-pembroke** [comment]
Fixed with #53
